core/src/main/scala/com/stripe/agate/tensor/Tensor.scala

```diff
@@ -229,7 +231,16 @@ abstract class Tensor[D <: DataType] {
           case Some(f) =>
             this.map(dest)(f)
           case None =>
-            throw new IllegalArgumentException(s"cannot cast from $dataType to $dest")
+            OnnxNumber.forDataType(dest) match {
+              case Success(num) if dataType == DataType.Bool =>
+                val f = num.fromLong(0L)
+                val t = num.fromLong(1L)
+                this.map(dest) { (x: Any) =>
```

Comment:
Huh: why is this `Any`?

Reply:
So, we've lost the type here. While we know that `dataType == Bool`, since we did the check on line 235, the Scala compiler doesn't know this. We could cast it back to a bool, but since functions are contravariant and scala will gladly let you compare `Any` with anything, we can cheat and skip the whole "convert to bool first" step and give a `map` a function that technically works with "any" type. Since we know it's bool, we just compare it against `true`. It's all kind of nasty - I'm going to change this and just use a cast, since it'll just do an `asInstanceOf` anyways...